tiledb/sm/filesystem/azure.cc

```diff
@@ -649,20 +751,211 @@ Status Azure::touch(const URI& uri) const {
   return Status::Ok();
 }
 
-Status Azure::write(const URI& uri, const void* buffer, uint64_t length) {
+Status Azure::write(
+    const URI& uri, const void* const buffer, const uint64_t length) {
+  if (!uri.is_azure()) {
+    return LOG_STATUS(Status::AzureError(
+        std::string("URI is not an Azure URI: " + uri.to_string())));
+  }
+
+  Buffer* const write_cache_buffer = get_write_cache_buffer(uri.to_string());
+
+  uint64_t nbytes_filled;
+  RETURN_NOT_OK(
+      fill_write_cache(write_cache_buffer, buffer, length, &nbytes_filled));
+
+  if (!use_block_list_upload_) {
+    if (nbytes_filled != length) {
+      std::stringstream errmsg;
+      errmsg << "Direct write failed! " << nbytes_filled
```

Comment:
is this error message backwards -- `length` should be bytes attempted and `nbytes_filled` the result?

Reply:
I think the last word of this error message can just be changed to "requested" instead of "resulted". This was a copy+paste from the S3 implementation so I'll change it there too.